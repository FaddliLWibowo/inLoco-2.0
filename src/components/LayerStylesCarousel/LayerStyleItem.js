import React from 'react'

const LayerStyleItem = ({layer, style, index, onStyleClick, onStyleMouseOver, onStyleMouseOut}) => {

    let itemClassName = 'layer-styles-carousel--list-item'
    if (layer && layer.selectedLayerStyleId === index) {
        itemClassName += ' selected'
    }

    function styleClick() {
        return onStyleClick(layer, index)
    }

    function styleMouseOver() {
        return onStyleMouseOver(layer, index)
    }

    function styleMouseOut() {
        return onStyleMouseOut(layer, index)
    }

    return (
        <li
            className={itemClassName}
            onClick={styleClick}
        >
            { style ?
                <img
                    className="layer-styles-carousel--image"
                    src={style.thumb}
                    alt={style.title}
                    onMouseOver={styleMouseOver}
                    onMouseOut={styleMouseOut}
                />
            : '' }
        </li>
    )
}

export default LayerStyleItem
