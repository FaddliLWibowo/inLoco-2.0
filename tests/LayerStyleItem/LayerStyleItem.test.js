import React from 'react';
import LayerStyleItem from '../../src/components/LayerStylesCarousel/LayerStyleItem.js';
import renderer from 'react-test-renderer';

it('LayerStyleItem component renders correctly with no data', () => {
    const component = shallow(<LayerStyleItem/>);
    expect(component).toMatchSnapshot();
});
