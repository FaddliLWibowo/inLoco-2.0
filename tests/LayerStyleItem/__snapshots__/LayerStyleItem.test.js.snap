// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerStyleItem component renders correctly with no data 1`] = `
<li
  className="layer-styles-carousel--list-item"
  onClick={[Function]}
/>
`;
